3-Solving-Problems-By-Searching/nodeExpansion.js

```diff
@@ -8,42 +8,59 @@ function nodeExpansionAgent(adjMatrix,initial){
   this.frontier = [initial];
 
   this.nodes = new Array(adjMatrix.length);
-  for(var i=0;i < this.nodes.length; i++){
+  for (var i = 0; i < this.nodes.length; i++) {
     this.nodes[i] = {
-      state : this.UNEXPLORED
+      state: this.UNEXPLORED,
+      cost: Number.POSITIVE_INFINITY,
+      parent: null
     };
   }
-  this.nodes[initial] = {state:this.EXPANDABLE}
-  this.getState = function(node){
+  this.nodes[initial] = {
+    state: this.EXPANDABLE,
+    cost: 0
+  };
+  this.getState = function(node) {
     return this.nodes[node].state;
   };
 
-  this.getNeighbors = function(node){
+  this.getNeighbors = function(node) {
     var neighbors = []
-    for(var i=0; i < this.adjMatrix[node].length ; i++){
-      if(this.adjMatrix[node][i] == 1){
+    for (var i = 0; i < this.adjMatrix[node].length; i++) {
+      if (this.adjMatrix[node][i] > 0) {
```

Comment:
I used a slightly different approach here. My adjacency matrix was initially designed for unweighted graphs and hence has 0s and 1s. Since I am reusing the code, I adjusted the 1s to represent the cost and 0 indicates no path between the nodes. This is because I assumed there won't be any edge with cost 0.